unstructured/ingest/v2/processes/connectors/salesforce.py

```diff
@@ -0,0 +1,293 @@
+    """There are no categories with that name."""
+
+
+CONNECTOR_TYPE = "salesforce"
+
+if TYPE_CHECKING:
+    pass
+
+SALESFORCE_API_VERSION = "57.0"
+
+ACCEPTED_CATEGORIES = ["Account", "Case", "Campaign", "EmailMessage", "Lead"]
+
+EMAIL_TEMPLATE = Template(
+    """MIME-Version: 1.0
+Date: $date
+Message-ID: $message_identifier
+Subject: $subject
+From: $from_email
+To: $to_email
+Content-Type: multipart/alternative; boundary="00000000000095c9b205eff92630"
+--00000000000095c9b205eff92630
+Content-Type: text/plain; charset="UTF-8"
+$textbody
+--00000000000095c9b205eff92630
+Content-Type: text/html; charset="UTF-8"
+$htmlbody
+--00000000000095c9b205eff92630--
+""",
+)
+
+
+@dataclass
+class SalesforceAccessConfig(AccessConfig):
+    consumer_key: str
+    private_key: str
+
+    @requires_dependencies(["cryptography"])
+    def get_private_key_value_and_type(self) -> t.Tuple[str, t.Type]:
+        from cryptography.hazmat.primitives import serialization
+
+        try:
+            serialization.load_pem_private_key(data=self.private_key.encode("utf-8"), password=None)
+        except ValueError:
+            pass
+        else:
+            return self.private_key, str
+
+        if Path(self.private_key).is_file():
+            return self.private_key, Path
+
+        raise ValueError("private_key does not contain PEM private key or path")
+
+
+@dataclass
+class SalesforceConnectionConfig(ConnectionConfig):
+    username: str
+    access_config: SalesforceAccessConfig = enhanced_field(sensitive=True)
+
+    @requires_dependencies(["simple_salesforce"], extras="salesforce")
+    def get_client(self):
+        from simple_salesforce import Salesforce
+
+        pkey_value, pkey_type = self.access_config.get_private_key_value_and_type()
+
+        return Salesforce(
+            username=self.username,
+            consumer_key=self.access_config.consumer_key,
+            privatekey_file=pkey_value if pkey_type is Path else None,
+            privatekey=pkey_value if pkey_type is str else None,
+            version=SALESFORCE_API_VERSION,
+        )
+
+
+@dataclass
+class SalesforceIndexerConfig(IndexerConfig):
+    categories: t.List[str]
+
+
+@dataclass
+class SalesforceIndexer(Indexer):
+    connection_config: SalesforceConnectionConfig
+    index_config: SalesforceIndexerConfig
+
+    @requires_dependencies(["simple_salesforce"], extras="salesforce")
+    def list_files(self) -> t.List[FileData]:
+        """Get Salesforce Ids for the records.
+        Send them to next phase where each doc gets downloaded into the
+        appropriate format for partitioning.
+        """
+        from simple_salesforce.exceptions import SalesforceMalformedRequest
+
+        client = self.connection_config.get_client()
+
+        files_list = []
+        for record_type in self.index_config.categories:
+            if record_type not in ACCEPTED_CATEGORIES:
+                raise ValueError(f"{record_type} not currently an accepted Salesforce category")
```

Comment:
Might not need this if the typing can run the check for us (comment above)

Reply:
@rbiseck3 how do we get the typing to activate for a standard user? If I run the example_salesforce.py it doesn't care if the category is legit or not without this.

Reply:
we determined that the typing was not worth it.